sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -161,7 +161,8 @@ abstract class Optimizer(catalogManager: CatalogManager)
     // LocalRelation and does not trigger many rules.
     Batch("LocalRelation early", fixedPoint,
       ConvertToLocalRelation,
-      PropagateEmptyRelation) ::
+      PropagateEmptyRelation,
+      UpdateAttributeNullability) ::
```

Comment:
Can you leave a comment above explaining why we run `UpdateAttributeNullability` here?

Reply:
Sure, added in the latest commit.